ngraph/frontend/paddlepaddle/src/model.cpp

```diff
@@ -0,0 +1,209 @@
+    std::string ext = ".pdmodel";            
+    std::string model_file(path);
+    if (path.length() >= ext.length() && (0 == path.compare(path.length() - ext.length(), ext.length(), ext)))
+    {
+        weights_composed = true;
+        auto weights_file = path.replace(path.size() - ext.size(), ext.size(), ".pdiparams");
+        weights_stream = std::ifstream(weights_file, std::ios::binary);
+        if (!weights_stream || !weights_stream.is_open())
+        {
+            std::cerr << "Model file cannot be opened" << std::endl;
+        }
+    } else {
+        weights_composed = false;
+        model_file += "/__model__";
+    }
+
+    std::ifstream pb_stream(model_file, std::ios::binary);
+    std::cout << "Model Parsed: " << fw_ptr->ParseFromIstream(&pb_stream) << std::endl;
+
+    std::cout << "Blocks number: " << fw_ptr->blocks().size() << std::endl;
+    for (const auto& block : fw_ptr->blocks()) {
+        var_places.push_back(std::map<std::string, std::shared_ptr<VarPlacePDPD>>());
+        for (int i = 0; i < block.vars().size(); i++) {
+            var_places.back()[block.vars()[i].name()] = std::make_shared<VarPlacePDPD>(VarPlacePDPD(&(block.vars()[i])));
+        }
+
+        op_places.push_back(std::vector<std::shared_ptr<OpPlacePDPD>>());
+        for (int i = 0; i < block.ops_size(); i++) {
+            auto& op = block.ops()[i];
+            auto op_place = std::make_shared<OpPlacePDPD>(OpPlacePDPD(&op));
+            for (const auto &output : op.outputs()) {
+                std::vector<std::weak_ptr<VarPlacePDPD>> out_vars;
+                for (auto& var_name : output.arguments()) {
+                    auto& var = var_places.back().at(var_name);
+                    var->producing_ops.push_back(op_place);
+                    out_vars.push_back(var);
+                }
+                op_place->outputs[output.parameter()] = out_vars;
+            }
+            std::map<std::string, google::protobuf::RepeatedPtrField<std::string>> inputs_dict;
+            for (const auto &input : op.inputs()) {
+                std::vector<std::weak_ptr<VarPlacePDPD>> in_vars;
+                for (auto& var_name : input.arguments()) {
+                    auto& var = var_places.back().at(var_name);
+                    var->consuming_ops.push_back(op_place);
+                    in_vars.push_back(var);
+                }
+                op_place->inputs[input.parameter()] = in_vars;
+            }
+            op_places.back().push_back(op_place);
+        }
+    }
+}
+
+template<typename T>
+std::vector<T> InputModelPDPD::InputModelPDPDImpl::getWeight(const std::string& name, int64_t tensor_length) {
+    std::vector<T> tensor_data(tensor_length, 0);
+
+    std::ifstream is;
+    std::ifstream* stream_ptr;
+    if (weights_composed) {
+        stream_ptr = &weights_stream;
+    } else {
+        is = std::ifstream(path + "/" + name, std::ios::in | std::ifstream::binary);
```

Comment:
it may be good to have platform-dependent file system separator, like FileUtils::FileSeparator in inference engine